kratos/utilities/nurbs_utilities/snake_sbm_utilities.h

```diff
@@ -0,0 +1,213 @@
+         * @param rEchoLevel Verbosity level for logging.
+         * @param knotVectorU Knot vector in the U-direction.
+         * @param knotVectorV Knot vector in the V-direction.
+         * @param mParameters Input parameters for the process.
+         *                      ex:
+         *                      {
+                                    "sbm_parameters": {
+                                        "lambda_inner": 0.5,
+                                        "lambda_outer": 0.5,
+                                        "number_of_inner_loops": 1
+                                    }
+                                }
+         * @param isInnerLoop Boolean flag to indicate if the loop is inner or outer.
+         */
+        static void CreateTheSnakeCoordinates(ModelPart& rIgaModelPart, 
+                                            ModelPart& rSkinModelPartInitial,
+                                            ModelPart& rSkinModelPart, 
+                                            int rEchoLevel, 
+                                            Vector& knotVectorU, 
+                                            Vector& knotVectorV, 
+                                            const Parameters mParameters,
+                                            bool isInnerLoop);
+
+        /**
+         * @brief Performs a single step in the snake algorithm for 2D models.
+         * @param rSkinModelPart The skin model part to be updated.
+         * @param knotSpansAvailable Knot spans available for the snake.
+         * @param idMatrixKnotSpansAvailable ID of the matrix tracking available knot spans.
+         * @param knotSpansUV Knot spans in UV coordinates.
+         * @param conditionCoords XY coordinates of control points.
+         * @param knotStepUV Step size in UV space.
+         * @param startingPositionUV Starting position in UV space.
+         */
+        static void SnakeStep(ModelPart& rSkinModelPart, 
+                            std::vector<std::vector<std::vector<int>>> &knotSpansAvailable, 
+                            int idMatrixKnotSpansAvailable, 
+                            std::vector<std::vector<int>> knotSpansUV, 
+                            std::vector<std::vector<double>> conditionCoords, 
+                            Vector knotStepUV, 
+                            Vector startingPositionUV);
+
+    private:
+        ///@name IGA functionalities
+        ///@{
+        using PointType = Node;
+        using PointTypePointer = Node::Pointer;
+        using PointVector = std::vector<PointType::Pointer>;
+        using PointIterator = std::vector<PointType::Pointer>::iterator;
+        using DistanceVector = std::vector<double>;
+        using DistanceIterator = std::vector<double>::iterator;
+        using DynamicBins = BinsDynamic<3, PointType, PointVector, PointTypePointer, PointIterator, DistanceIterator>;
+        using PointerType = DynamicBins::PointerType;
```

Comment:
```suggestion
        using DynamicBinsPointerType = DynamicBins::PointerType;
```
PointerType is meaningless when browsing the code.